lib/channels/src/index.ts

```diff
@@ -0,0 +1,152 @@
+export type ChannelHandler<TEventArgs = any> = (event: ChannelEvent<TEventArgs>) => void;
+
+export interface ChannelTransport<TEventArgs = any> {
+  send(event: ChannelEvent<TEventArgs>): void;
+  setHandler(handler: ChannelHandler<TEventArgs>): void;
+}
+
+export interface ChannelEvent<TEventArgs = any> {
+  type: string; // eventName
+  from: string;
+  args: TEventArgs;
+}
+
+export interface Listener<TEventArgs = any> {
+  (...args: TEventArgs[]): void;
+
+  ignorePeer?: boolean;
+}
+
+interface EventsKeyValue {
+  [key: string]: Listener[];
```

Comment:
What does this `[key: string]` part do? 😮

Reply:
🆗

Reply:
Just creating a key-value type

https://www.typescriptlang.org/play/#src=interface%20KeyVal%20%7B%0D%0A%20%20%20%20%5Bkey%3A%20string%5D%3A%20number%3B%0D%0A%7D%0D%0A%0D%0Aconst%20keyValue%3A%20KeyVal%20%3D%20%7B%0D%0A%20%20%20%20someKey%3A%206%2C%0D%0A%20%20%20%20anotherKey%3A%2010%2C%0D%0A%20%20%20%20notPossible%3A%20%22foo%22%0D%0A%7D

key obviously is a string but the important part is the type of `value`